pkg/controller/deployment_inplace.go

```diff
@@ -193,6 +194,18 @@ func (dc *controller) syncMachineSets(ctx context.Context, oldMachineSets []*v1a
 		// uncordon the node since the inplace update is successful.
 		node.Spec.Unschedulable = false
 
+		// remove the PreferNoSchedule taint if it exists which was added during the inplace update.
+		node.Spec.Taints = slices.DeleteFunc(node.Spec.Taints, func(t v1.Taint) bool {
+			return t.Key == PreferNoScheduleKey && t.Value == "True" && t.Effect == v1.TaintEffectPreferNoSchedule
+		})
```

Comment:
Why is this taint not required anymore?

Reply:
This is added at the start of the in-place update. The machine that will go through the update, pod should preferably not be scheduled on that node.
Later, after the update is successful we need to remove it from the node